### PR TITLE
Add GapConductanceConstant and associated Action changes

### PR DESCRIPTION
**Pull request by bwspenc**
ref #13061

### CODE REVIEW

**Comment by tophmatthews** [comment]
So should we use this instead for highly conductive gaps? I think this assumes away contact conduction, but I imagine that should be okay?

In other words...what should I watch for if I implement this instead of the old models?

File: modules/heat_conduction/include/materials/GapConductanceConstant.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#include "Material.h"
+
+class GapConductanceConstant : public Material
+{
+public:
+  GapConductanceConstant(const InputParameters & parameters);
+
+  static InputParameters actionParameters();
+
+  virtual ~GapConductanceConstant() {}
```
Comment (permcody) [inline modules/heat_conduction/include/materials/GapConductanceConstant.h]: Note: You don't need to define an empty destructor in inherited classes where one of the bases has already declared a virtual destructor. You can simply remove this line or use the C++11 syntax to explicitly `delete` this destructor.

**Review by permcody — DISMISSED** [review]

**Comment by bwspenc** [comment]
@tophmatthews That's exactly what this is intended for. Basically this is just enforcing that the temperatures are equal on the two sides of the gap, but with a penalty parameter that can be understood as the conductance.  The conductance would no longer be a function of the gap size. This is just a more elegant version of what Al accomplished by setting min_gap really big. I expect that this will be what you guys want to use on your problems because the conductance is high enough that its effect on the solution is negligible.

File: modules/heat_conduction/include/materials/GapConductanceConstant.h (hunk 1 above)
Comment (bwspenc) [reply modules/heat_conduction/include/materials/GapConductanceConstant.h]: Thanks for pointing that out.  I just pushed an updated version that removes that.

**Comment by bwspenc** [comment]
@permcody I see you assigned me, but this is my PR...

**Comment by permcody** [comment]
Oops - ok

**Review by dschwen — APPROVED** [review]
:+1:

**Pull request merged**